File: app/src/main/java/com/patloew/template/ui/main/MainViewModel.java
package com.patloew.template.ui.main;

import com.patloew.template.injection.scopes.PerActivity;
import com.patloew.template.ui.base.viewmodel.BaseViewModel;

import javax.inject.Inject;

@PerActivity
public class MainViewModel extends BaseViewModel<MainMvvm.View> implements MainMvvm.ViewModel {

    @Inject
    public MainViewModel() { }

}

File: app/src/main/java/com/patloew/template/ui/main/MainViewModel.kt
package com.patloew.template.ui.main

import com.patloew.template.injection.scopes.PerActivity
import com.patloew.template.ui.base.viewmodel.BaseViewModel

import javax.inject.Inject

@PerActivity
class MainViewModel
@Inject
constructor() : BaseViewModel<MainMvvm.View>(), MainMvvm.ViewModel
